src/views/HomeView.vue

```diff
@@ -19,8 +19,11 @@ const panelStore = usePanelsStore()
     <div class="z-10 absolute inset-x-0 bottom-0 max-w-max m-auto">
       <PhotoGallery></PhotoGallery>
     </div>
+    <!--Do lazy loading for planning view.
+      Only render it once, and after that 'hide' the planning view so that we can keep its state and no need to load it again. -->
     <div
       class="absolute h-screen w-screen"
+      v-if="panelStore.hasPlanningViewRendered"
```

Comment:
I may be wrong, but isn't it rather the opposite? Hide the map with the z-index until it is rendered so rather condition the z-index with the hasPlanningViewRendered

Reply:
This v-if here is to prevent unnecessary rendering before the PlanningView is rendered

So here is the case:
- The user opens the home page for the first time (`hasPlanningViewRendered` is False by default), so the planning view is not rendered. 
- User clicks the button to open the planning view -> `hasPlanningViewRendered` is set to True -> the planning view is rendered. In this case, the `isPlanningViewShown` is set to True -> the planning view has z-20, so the user can see it now.
- User presses the back button, `isPlanningViewShown` is set to False -> the planning view has -z-20, so the user can NOT see it now. The `hasPlanningViewRendered` is still True, so the state is kept.